driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -621,9 +649,61 @@ private static void updateLocationInfo(
     if (!isInitialConnection) cluster.loadBalancingPolicy().onAdd(host);
   }
 
-  private static void refreshNodeListAndTokenMap(
-      Connection connection,
-      Cluster.Manager cluster,
+  /**
+   * Resolves peering information by doing the following:
+   *
+   * <ol>
+   *   <li>if <code>isPeersV2</code> is true, query the <code>system.peers_v2</code> table,
+   *       otherwise query <code>system.peers</code>.
+   *   <li>if <code>system.peers_v2</code> query fails, set <code>isPeersV2</code> to false and call
+   *       selectPeersFuture again.
+   * </ol>
+   *
+   * @param connection connection to send request on.
+   * @return result of peers query.
+   */
+  private ListenableFuture<ResultSet> selectPeersFuture(final Connection connection) {
+    if (isPeersV2) {
```

Comment:
For < C* 4.x, maybe we should just not even try the peers_v2 table.  We could only allow this for protocol V5+.  I think that's an ok limitation, and we can add some kind of way to bypass the V5 requirement for our Scassandra tests.

Reply:
Thinking about this more. I think for 3.x, this should be more of an opt-in thing via a builder option.  For a minor version making an extra query on initialization seems kinda iffy, and I think it's ok to require use of this feature to be explicit, since the 3.x driver is more driven around ip addresses and global setting of port (via `withPort`), so we should probably make a separate option so users can opt-in to this capability.

Reply:
We discussed this more and decided that it's ok to try `peers_v2` table first.